OpenRA.Mods.Common/Activities/Parachute.cs

```diff
@@ -0,0 +1,77 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class Parachute : Activity
+	{
+		WPos currentPosition;
+		WVec fallVector;
+
+		public Parachute(Actor self, WPos dropPosition)
+		{
+			var um = self.TraitOrDefault<UpgradeManager>();
```

Comment:
In general, activities may be queued and not run until an arbitrary time after they are constructed.  It is therefore best practice to not do anything that changes world state from their constructor.  You can call these from the first `Tick()`.